scripts/create-instance.sh

```diff
@@ -0,0 +1,391 @@
+ci_user_username=$(cf service-key "${service_name_cf_ci_user}" "${service_name_cf_ci_user}-key" | tail -n +2 | jq -r '.username')
+##
+
+##
+# Pages!
+##
+if [[ "$product" = "pages" ]]; then
+  echo "Configuring services for Pages"
+
+  # Configure UAA 
+  uaa_service_creds="$(cat <<- EOF
+    {
+      "authorizationURL": "https://login.${UAA_DOMAIN}/oauth/authorize",
+      "clientID": "${UAA_CLIENT_ID}",
+      "clientSecret": "${UAA_CLIENT_SECRET}",
+      "logoutURL": "https://uaa.${UAA_DOMAIN}/logout.do",
+      "tokenURL": "https://uaa.${UAA_DOMAIN}/oauth/token",
+      "userURL": "https://uaa.${UAA_DOMAIN}/userinfo"
+    }
+EOF
+)"
+
+  # Create Mailer creds and network policy
+  cf target -s email -o "${cf_org}"
+  auth_service_guid=$(cf curl '/v3/service_instances?names=auth' | jq -r '.resources[0].guid')
+  mailer_password=$(cf curl "/v3/service_instances/${auth_service_guid}/credentials" | jq -r '.password')
+  mailer_username=$(cf curl "/v3/service_instances/${auth_service_guid}/credentials" | jq -r '.username')
+
+  cf target -s "${cf_space}" -o "${cf_org}"
+
+  mailer_creds="$(cat <<- EOF
+    {
+      "host": "pages-mailer.apps.internal:8000",
+      "password": "${mailer_password}",
+      "username": "${mailer_username}"
+    }
+EOF
+)"
+
+  slack_creds="$(cat <<- EOF
+    {
+      "url": "${SLACK_URL}"
+    }
+EOF
+)"
+
+  # Create Network Policy
+  cf add-network-policy $web_app_name pages-mailer -s email -o ${org_name}
+else
+  uaa_service_creds="$(cat <<- EOF
+    {
+      "authorizationURL": "",
+      "clientID": "",
+      "clientSecret": "",
+      "logoutURL": "",
+      "tokenURL": "",
+      "userURL": ""
+    }
+EOF
+)"
+
+  mailer_creds="$(cat <<- EOF
+    {
+      "host": "",
+      "password": "",
+      "username": ""
+    }
+EOF
+)"
+
+  slack_creds="$(cat <<- EOF
+    {
+      "url": ""
+    }
+EOF
+)"
+fi
+
+cf create-user-provided-service "${service_name_uaa}" -p "${uaa_service_creds}"
+cf create-user-provided-service mailer -p "${mailer_creds}"
+cf create-user-provided-service slack -p "${slack_creds}"
+##
+
+## All done!!
+echo "Infrastructure created for ${cf_space}!!"
+echo
+echo
+echo "Please finish the configuration by doing the following:"
+echo
+echo "-  In the ${product}-proxy-${env_type} ${cf_space} vars file, please set 'shared-bucket=${shared_bucket}'"
+echo
+echo "-  In the Concourse jumpbox, please run the following:"
+echo "     credhub set -n /concourse/pages/${cf_space}-cf-password -t value -v ${ci_user_password}"
+echo "     credhub set -n /concourse/pages/${cf_space}-cf-username -t value -v ${ci_user_username}"
+echo
+echo "-  Add any additional humans:"
+echo "     cf set-space-role <username@gsa.gov> ${cf_org} ${cf_space} SpaceDeveloper"
+echo "     cf set-space-role <username@gsa.gov> ${cf_org} ${cf_sites_space} SpaceDeveloper"
+echo
+echo "You may now deploy the pages and applications:"
```

Comment:
Be more explicit about kicking off a build in CI?

Reply:
What do you mean?

Reply:
If we want any more documentation on using Concourse or Circle to kick off a deploy. But it's not like we're going to use this that offten.